scio-core/src/main/scala/com/spotify/scio/values/SCollection.scala

```diff
@@ -806,15 +807,15 @@ sealed trait SCollection[T] extends PCollectionWrapper[T] {
     if (context.isTest) {
       saveAsInMemoryTap
     } else {
+      val elemCoder = this.getCoder[T]
       this
         .parDo(new DoFn[T, GenericRecord] {
-          private val coder = KryoAtomicCoder[T]
           override def processElement(c: DoFn[T, GenericRecord]#ProcessContext): Unit = {
-            c.output(AvroBytesUtil.encode(coder, c.element()))
+            c.output(AvroBytesUtil.encode(elemCoder, c.element()))
           }
         })
         .saveAsAvroFile(path, numShards, AvroBytesUtil.schema)
-      context.makeFuture(ObjectFileTap[T](path + "/part-*"))
+      context.makeFuture(ObjectFileTap[T](path + "/part-*", elemCoder))
```

Comment:
You don't need the optional arg `elemCoder` here since `ObjectFileTap` can figure out itself?